shared/actions/devices.js

```diff
@@ -103,11 +112,20 @@ export function removeDevice (deviceID: string, name: string, currentDevice: boo
       revokeRevokeDeviceRpc({
         param: {deviceID, force: false},
         callback: error => {
-          dispatch({
-            type: Constants.deviceRemoved,
-            payload: error,
-            error: !!error,
-          })
+          if (error) {
+            dispatch({
+              type: Constants.deviceRemoved,
+              payload: error,
+              error: true,
+            })
+          } else {
+            dispatch({
+              type: Constants.deviceRemoved,
+              payload: error,
+              error: false,
```

Comment:
having said that i have a longstanding request to make payload not-null if its not an error. its annoying to have to be like 

```if (action.payload && action.payload.blah)```

all over the place